File: Java/opentraceability/src/main/java/opentraceability/gdst/events/GDSTDisaggregationEvent.java
package opentraceability.gdst.events;

import opentraceability.Constants;
import opentraceability.models.events.AggregationEventBase;
import opentraceability.models.events.EventAction;
import opentraceability.models.identifiers.PGLN;
import opentraceability.utility.attributes.OpenTraceabilityAttribute;
import opentraceability.utility.attributes.OpenTraceabilityJsonAttribute;

import java.net.URI;

public class GDSTDisaggregationEvent extends AggregationEventBase<GDSTILMD> implements IGDSTEvent {
    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "productOwner")
    @OpenTraceabilityJsonAttribute(name = "gdst:productOwner")
    private PGLN productOwner;

    @Override
    public PGLN getProductOwner() {
        return this.productOwner;
    }


    // Add constructor, getters, and setters here
    public GDSTDisaggregationEvent()
    {
        this.businessStep = URI.create("urn:epcglobal:cbv:bizstep:unpacking");
        this.action = EventAction.DELETE;
    }
}

File: CSharp/OpenTraceability.GDST/Events/GDSTDisaggregationEvent.cs
using OpenTraceability.Models.Events;
using OpenTraceability.Models.Identifiers;
using OpenTraceability.Utility.Attributes;

namespace OpenTraceability.GDST.Events
{
    public class GDSTDisaggregationEvent : AggregationEvent<GDSTILMD>, IGDSTProductOwnerEvent
    {
        [OpenTraceabilityJson("gdst:productOwner")]
        public PGLN? ProductOwner { get; set; }

        public GDSTDisaggregationEvent()
        {
            BusinessStep = new Uri("urn:epcglobal:cbv:bizstep:unpacking");
            Action = EventAction.DELETE;
        }
    }
}
